### PR TITLE
⚡️ [FEAT] : Integrate Trustchain logic in LedgerSync LLD - V1

### PR DESCRIPTION
**Pull request by mcayuelas-ledger**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - Trustchain Store is persistent
  - Integration of Trustchain
    - Add Member
    - Remove Member
    - Destroy Trustchain

### 📝 Description

Integrates the trustchain lib and store integration on the actual UI of the trustchain flows

- Arch reworked
- Hooks dedicated

### ❓ Context

- **JIRA or GitHub link**: [LIVE-12631] 

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-12631]: https://ledgerhq.atlassian.net/browse/LIVE-12631?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/newArch/WalletSync/Flows/Activation/03-ActivationOrSynchroWithTrustchain.tsx [hunk 1]
```diff
@@ -2,26 +2,52 @@ import React, { useCallback, useEffect } from "react";
...
+      }),
+    );
+  }, [sdk, dispatch, device?.deviceId]);
+
   useEffect(() => {
-    setTimeout(() => {
-      !hasError && stuffHandledByTrustchain();
-    }, 3000);
-  }, [hasError, stuffHandledByTrustchain]);
+    !hasError && stuffHandledByTrustchain();
```
Comment (gre) [inline apps/ledger-live-desktop/src/newArch/WalletSync/Flows/Activation/03-ActivationOrSynchroWithTrustchain.tsx:48]: improvements i see:
- we need to handle possible error thrown by stuffHandledByTrustchain and display them to user (using the Generic Error rendering which should tell the user the precise error instead of the UnsecuredError )
- Q: is there a way, as a user, to interrupt the flow? because we don't handle the unsubscribe of these, and it's possible that runWithDevice never ends which can introduce race conditions

File: apps/ledger-live-desktop/src/newArch/WalletSync/Flows/ManageBackup/useDeleteData.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { Flow, Step } from "~/renderer/reducers/walletSync";
+
+export function useDeleteData() {
+  const dispatch = useDispatch();
+  const sdk = useTrustchainSdk();
+  const trustchain = useSelector(trustchainSelector);
+  const memberCredentials = useSelector(memberCredentialsSelector);
+
+  if (!trustchain || !memberCredentials) {
+    throw new Error("trustchain or memberCredentials is missing");
```
Comment (gre) [inline apps/ledger-live-desktop/src/newArch/WalletSync/Flows/ManageBackup/useDeleteData.ts:20]: dangerous to throw in a hook. 
to me, instead we should return a callback that will do nothing if trustchain is falsy, because in that case there will be no more things to delete.

File: apps/ledger-live-desktop/src/newArch/WalletSync/Flows/Synchronize/02-QRCodeStep.tsx [hunk 3]
```diff
@@ -35,12 +43,54 @@ export default function SynchWithQRCodeStep({ displayPinCode }: Props) {
...
+
+  const onStart = useCallback(() => {
+    if (!trustchain || !memberCredentials) return;
+    setError(null);
+    createQRCodeHostInstance({
+      onDisplayQRCode: url => {
+        setUrl(url);
+      },
+      onDisplayDigits: digits => {
+        dispatch(setQrCodePinCode(digits)); // Should we assert that it is a string of 3 digits ?
```
Comment (gre) [inline apps/ledger-live-desktop/src/newArch/WalletSync/Flows/Synchronize/02-QRCodeStep.tsx:68]: actually today it's "3" but it could be something else, this is a config in the protocol of QR Code, it could be changed in future and UI just need to adapt to the number of digits ideally: https://github.com/LedgerHQ/ledger-live/blob/develop/libs/trustchain/src/qrcode/index.ts#L155

File: apps/ledger-live-desktop/src/newArch/WalletSync/useTrustchainSdk.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+const defaultContext = { applicationId: 16, name: "Ledger Sync LLD" }; // TODO : get name dynamically depending on the platform
+
+export function useLiveAuthenticate() {
+  const sdk = useTrustchainSdk();
+  const trustchain = useSelector(trustchainSelector);
+  const memberCredentials = useSelector(memberCredentialsSelector);
+
+  if (!trustchain || !memberCredentials) {
```
Comment (gre) [inline apps/ledger-live-desktop/src/newArch/WalletSync/useTrustchainSdk.ts:33]: i think we should be able to useLiveAuthenticate with these falsy, it hsouldn't throw from the hook.
there will be error cases where we will need to reset the trustchain (e.g. when member is ejected)

**Review by gre — COMMENTED** [review]
some feedbacks

File: apps/ledger-live-desktop/src/newArch/WalletSync/useTrustchainSdk.ts (hunk 4 above)
Comment (mcayuelas-ledger) [reply apps/ledger-live-desktop/src/newArch/WalletSync/useTrustchainSdk.ts:33]: as `auth(trustchain: Trustchain, memberCredentials: MemberCredentials): Promise<JWT>;
`
We can't :/

File: libs/trustchain/src/qrcode/index.ts [hunk 5]
```diff
@@ -34,7 +34,7 @@ export async function createQRCodeHostInstance({
 }): Promise<void> {
   const ephemeralKey = await crypto.randomKeypair();
   const publisher = crypto.to_hex(ephemeralKey.publicKey);
-  const url = `${getEnv("TRUSTCHAIN_API").replace("http", "ws")}/v1/qr?host=${publisher}`;
+  const url = `${getEnv("TRUSTCHAIN_API")}/v1/qr?host=${publisher}`.replace(/^http/, "ws");
```
Comment (gre) [inline libs/trustchain/src/qrcode/index.ts:37]: can you undo this change? the develop's version is correct

File: libs/trustchain/src/store.ts [hunk 6]
```diff
@@ -4,30 +4,36 @@
...
   trustchain: Trustchain | null;
   memberCredentials: MemberCredentials | null;
+  jwt: JWT | null;
 };
 
 export const getInitialStore = (): TrustchainStore => {
   return {
     trustchain: null,
     memberCredentials: null,
+    jwt: null,
```
Comment (gre) [inline libs/trustchain/src/store.ts:19]: we should undo this too, no need to store the jwt anymore

File: apps/ledger-live-desktop/src/renderer/storage.ts [hunk 7]
```diff
@@ -52,6 +55,7 @@ type DatabaseValues = {
   countervalues: Countervalues;
   postOnboarding: PostOnboarding;
   settings: Settings;
+  trustchainStore: TrustchainStore;
```
Comment (gre) [inline apps/ledger-live-desktop/src/renderer/storage.ts:58]: correct me if i'm wrong but i think we didn't implemented the encryption of the trustchainStore yet? 
i can help on this part

**Comment by gre** [comment]
+ don't forget to update & document the PR description

File: apps/ledger-live-desktop/src/renderer/storage.ts (hunk 7 above)
Comment (mcayuelas-ledger) [reply apps/ledger-live-desktop/src/renderer/storage.ts:58]: Not yet

File: apps/ledger-live-desktop/src/newArch/WalletSync/hooks/useTrustchainSdk.ts [hunk 8]
```diff
@@ -13,10 +13,21 @@ export function runWithDevice<T>(
   return lastValueFrom(withDevice(deviceId || "")(transport => from(fn(transport))));
 }
 
-const defaultContext = { applicationId: 16, name: `LLD Sync ${randomUUID().slice(0, 8)}` }; // TODO : get name dynamically depending on the platform
+const platformMap: Record<string, string | undefined> = {
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-desktop/src/newArch/WalletSync/hooks/useTrustchainSdk.ts:16]: Why undefined?

Comment (gre) [reply apps/ledger-live-desktop/src/newArch/WalletSync/hooks/useTrustchainSdk.ts:16]: `Record<string, F>` is rarely safe, we always want `Record<string, F | undefined>`

here is an example

Capture d’écran 2024-07-08 à 16 33 13


in our case, os.platform() can theorically be something else than linux,mac,windows.

File: apps/ledger-live-desktop/src/newArch/WalletSync/Flows/Activation/03-ActivationOrSynchroWithTrustchain.tsx [hunk 9]
```diff
@@ -1,31 +1,44 @@
...
-  return hasError ? (
-    <UnsecuredError />
-  ) : (
-    <Loading title={t("walletSync.loading.title")} subtitle={t("walletSync.loading.activation")} />
-  );
+  if (device) {
+    return addMemberMutation.isError ? (
+      <ErrorDisplay error={addMemberMutation.error} withExportLogs onRetry={onRetry} />
+    ) : (
+      <FollowStepsOnDevice modelId={device.modelId as DeviceModelId} />
```
Comment (LucasWerey) [inline apps/ledger-live-desktop/src/newArch/WalletSync/Flows/Activation/03-ActivationOrSynchroWithTrustchain.tsx:34]:
```suggestion
      <FollowStepsOnDevice modelId={device.modelId} />
```

Comment (LucasWerey) [reply apps/ledger-live-desktop/src/newArch/WalletSync/Flows/Activation/03-ActivationOrSynchroWithTrustchain.tsx:34]: And the import

File: apps/ledger-live-desktop/src/newArch/WalletSync/Flows/ManageInstances/01-ManageInstancesStep.tsx [hunk 10]
```diff
@@ -1,42 +1,77 @@
...
+            cta={t("walletSync.manageInstances.remove")}
+            onClick={
+              memberCredentials?.pubkey == instance.id
+                ? handleAutoRemove
+                : () => handleGoDeleteInstance(instance)
+            }
+            currentInstance={memberCredentials?.pubkey == instance.id}
+          />
+        ))}
+      </List>
```
Comment (LucasWerey) [inline apps/ledger-live-desktop/src/newArch/WalletSync/Flows/ManageInstances/01-ManageInstancesStep.tsx:64]: There is some ts errors here

**Review by gre — APPROVED** [review]

**Review by LucasWerey — APPROVED** [review]
